File: backend/src/main/java/com/classroom/backend/dto/auth/AuthResponse.java
package com.classroom.backend.dto.auth;

import com.classroom.backend.model.enums.UserRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

@Data
@AllArgsConstructor
@Builder
public class AuthResponse {
    private String token;
    private String id;
    private String name;
    private String email;
    private UserRole role;
    /** Short portal login id when different from contact email. */
    private String loginId;
    /** Set when registering a school in one step. */
    private String schoolId;
    private Boolean emailVerified;
    /** First portal login: user must choose a password before accessing the app. */
    private Boolean passwordSetupRequired;
    private String setupToken;
}

File: lib/models/auth_response.dart
class AuthResponse {
  const AuthResponse({
    required this.token,
    required this.id,
    required this.name,
    required this.email,
    required this.role,
    this.loginId,
  });

  final String token;
  final String id;
  final String name;
  final String email;
  final String role;
  final String? loginId;

  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      token: json['token'] as String,
      id: json['id'] as String,
      name: json['name'] as String? ?? '',
      email: json['email'] as String,
      role: json['role'] as String,
      loginId: json['loginId'] as String?,
    );
  }
}
